tensorflow/python/framework/tensor_shape.py

```diff
@@ -245,7 +245,10 @@ def is_compatible_with(self, other):
     Returns:
       True if this Dimension and `other` are compatible.
     """
-    other = as_dimension(other)
+    try:
+      other = as_dimension(other)
+    except (TypeError, ValueError) as e:
+      return (type(e), '::', e)
```

Comment:
I think this is even more confusing and also break the contract of returning a Boolean value. Now its returning a tuple with errors. 

And also, catching the error and rethrow it is kind of anti-pattern in the code. Why do u want to specially handle these errors? Is it just for hiding the stacktrace?